execution.go

```diff
@@ -892,10 +895,12 @@ func mergeMaps(dst, src map[string]interface{}) {
 
 			switch value := v.(type) {
 			case json.RawMessage:
-				var m map[string]interface{}
+				// we want to unmarshal only what's necessary, so unmarshal only
```

Comment:
Is the `RawMessage` always going to be an Object? It could also be an Array couldn't it

Reply:
This function is only called when merging root queries together and I can't think of a case where we'd need to unmarshal an array.
If we need to merge arrays it will be for boundary types and thus will be done in `executeChildStep`.
So I think this one should be safe.